@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,439 @@
+        maxRSq = Math.max(
+            maxRSq,
+            center.distanceToSquared(vertex.fromBufferAttribute(pos, indices[i])),
+            center.distanceToSquared(vertex.fromBufferAttribute(pos, indices[i + VERTEX_STRIDE]))
+        );
+    }
+    sphere.radius = Math.sqrt(maxRSq);
+    return sphere;
+}
+
+/**
+ * Finds the intersection of a ray with a extruded line.
+ * @param ray Intersection ray in object's local space.
+ * @param line The centerline.
+ * @param vExtrusion Line extrusion vector.
+ * @param normal Extrusion plane normal.
+ * @param hWidth Extrusion half width.
+ * @returns Distance of the extruded line intersection to the ray origin.
+ */
+function intersectExtrudedLine(
+    ray: THREE.Ray,
+    line: THREE.Line3,
+    vExtrusion: THREE.Vector3,
+    normal: THREE.Vector3,
+    hWidth: number
+): number {
+    const obb = tmpOBB;
+    line.getCenter(obb.position);
+    line.delta(obb.xAxis).normalize();
+    obb.yAxis.copy(vExtrusion);
+    obb.zAxis.copy(normal);
+    obb.extents.set(line.distance() / 2, hWidth, hWidth);
+
+    if (obb.contains(ray.origin)) {
+        return 0;
+    }
+    return obb.intersectsRay(ray) ?? Infinity;
+}
+
+/**
+ * Finds the intersection of a ray with the closest end cap of a extruded line.
+ * @param ray Intersection ray in object's local space.
+ * @param line The centerline.
+ * @param hWidth Extrusion half width.
+ * @returns Distance of the end cap intersection to the ray origin.
+ */
+function intersectClosestEndCap(ray: THREE.Ray, line: THREE.Line3, hWidth: number): number {
+    const sphere = new THREE.Sphere(line.start, hWidth);
+    const startCapT = sphere.containsPoint(ray.origin)
+        ? 0
+        : ray.intersectSphere(sphere, tmpV4)
+        ? tmpV4.sub(ray.origin).length()
+        : Infinity;
+    sphere.center.copy(line.end);
+    const endCapT = sphere.containsPoint(ray.origin)
+        ? 0
+        : ray.intersectSphere(sphere, tmpV4)
+        ? tmpV4.sub(ray.origin).length()
+        : Infinity;
+    return Math.min(startCapT, endCapT);
+}
+
+/**
+ * Intersects line
+ * @param ray Intersection ray in object's local space.
+ * @param line The line to intersect.
+ * @param vExtrusion Line extrusion vector.
+ * @param hWidth The line's extrusion half width.
+ * @param hWidthSq The line's extrusion half width squared.
+ * @param plane The extrusion plane.
+ * @param interPlane The intersection of the ray with the extrusion plane.
+ * @param outInterLine The ray intersetion with the extruded line.
+ * @returns true if ray intersects the extruded line, false otherwise.
+ */
+function intersectLine(
+    ray: THREE.Ray,
+    line: THREE.Line3,
+    vExtrusion: THREE.Vector3,
+    hWidth: number,
+    hWidthSq: number,
+    plane: THREE.Plane,
+    interPlane: THREE.Vector3,
+    outInterLine: THREE.Vector3
+): boolean {
+    if (interPlane.equals(ray.origin) && ray.direction.dot(plane.normal) === 0) {
```

Comment:
Minor. I am little bit worried about lack of epsilon here, hitting exact ZERO may be hard to impossible in real life examples.

Reply:
I think we better discuss this on a call.